src/leap/bitmask/core/configurable.py

```
@@ -0,0 +1,262 @@
+
+
+class MissingConfigEntry(Exception):
+    """
+    A required config entry was not found.
+    """
+
+
+class ConfigurableService(service.MultiService):
+
+    config_file = u"bitmaskd.cfg"
+    service_names = ('mail', 'eip', 'zmq', 'web')
+
+    def __init__(self, basedir='~/.config/leap'):
+        service.MultiService.__init__(self)
+
+        path = os.path.abspath(os.path.expanduser(basedir))
+        if not os.path.isdir(path):
+            files.mkdir_p(path)
+        self.basedir = path
+
+        # creates self.config
+        self.read_config()
+
+    def get_config(self, section, option, default=None, boolean=False):
+        try:
+            if boolean:
+                return self.config.getboolean(section, option)
+
+            item = self.config.get(section, option)
+            return item
+
+        except (ConfigParser.NoOptionError, ConfigParser.NoSectionError):
+            if default is None:
+                fn = os.path.join(self.basedir, self.config_file)
+                raise MissingConfigEntry("%s is missing the [%s]%s entry"
+                                         % (_quote_output(fn),
+                                            section, option))
+            return default
+
+    def set_config(self, section, option, value):
+        if not self.config.has_section(section):
+            self.config.add_section(section)
+        self.config.set(section, option, value)
+        self.save_config()
+        self.read_config()
+        assert self.config.get(section, option) == value
+
+    def read_config(self):
+        self.config = ConfigParser.SafeConfigParser()
+        bitmaskd_cfg = self._get_config_path()
+
+        if not os.path.isfile(bitmaskd_cfg):
+            self._create_default_config(bitmaskd_cfg)
+
+        try:
+            with open(bitmaskd_cfg, "rb") as f:
+                self.config.readfp(f)
+        except EnvironmentError:
+            if os.path.exists(bitmaskd_cfg):
+                raise
```

you're right, it doesn't make sense since i'm creating the file just above.
I just copied the idiom from some other service and then modified.
btw, EnvironmentError is parent for IOError.

about the rant... yes, it's python. still, I think it's better to be precise regarding what you catch. and let it crash!